zulipterminal/ui_tools/buttons.py

```diff
@@ -288,11 +288,36 @@ def handle_link(self, *_: Any) -> None:
         """
         if self.is_internal_link:
             if '/#narrow/' in self.link:
-                # TODO: Handle internal link.
-                pass
+                self.handle_narrow()
             elif '/user_uploads/' in self.link:
                 # TODO: Open media in the native viewer.
                 pass
         else:
             # TODO: Open external link in the web browser.
             pass
+
+    def handle_narrow(self) -> None:
+        """
+        Handles stream narrow.
+        """
+        # TODO: Handle topic narrow.
+        link_split = self.link.split('/')
+
+        self.stream_id = int(link_split[5].split('-')[0])
+
+        # Return if the concerned stream is unsubscribed.
+        if self.stream_id not in self.model.stream_dict:
+            text = ' You are not subscribed to the stream'
+            self.view.set_footer_text(text, duration=3)
+            return
+
+        self.stream_name = self.model.stream_dict[self.stream_id]['name']
```

Comment:
Is this and `self.stream_id` for `narrow_to_stream`?

If so, it may be worth adding a comment with the assert statements.